### ISSUE TITLE
High cpu usage when moving windows.

### ISSUE THREAD

**Issue by doubledogdare610** [linuxmint/Cinnamon#4799]
Just as the title says. Cinnamon uses high cpu whenever I move windows. It is using 45-85%.

**Comment by JosephMcc** [comment]
Could you provide a bit more info? Cinnamon version, video card, drivers, etc. Does this happen with all windows or just certain ones? Might be a dumb question but you aren't running in software rendering mode are you?

**Comment by hamaryns** [comment]
I experience this also, since a few days.  Probably caused by some update or install.  

top - 09:33:32 up 46 min,  2 users,  load average: 0,99, 0,91, 0,71
Tasks: 370 total,   1 running, 369 sleeping,   0 stopped,   0 zombie
%Cpu(s): 17,5 us,  0,5 sy,  0,0 ni, 81,9 id,  0,1 wa,  0,0 hi,  0,0 si,  0,0 st
KiB Mem:   7853496 total,  5163960 used,  2689536 free,   346496 buffers
KiB Swap:  8058876 total,        0 used,  8058876 free.  2457188 cached Mem

  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND     
 2679 hendrik   20   0 1647264 213960  78960 S  64,0  2,7  18:14.07 cinnamon    
 2690 hendrik   20   0 1559124 705140  91912 S   2,7  9,0   1:59.28 firefox     
 2692 hendrik   20   0  724548 178468  72884 S   2,3  2,3   1:04.29 skype       

I often have to click a window twice to activate it.
$ uname -mrs
Linux 3.16.0-38-generic x86_64
$ lsb_release -a
No LSB modules are available.
Distributor ID:	LinuxMint
Description:	Linux Mint 17.2 Rafaela
Release:	17.2
Codename:	rafaela

Cinnamon 2.6.13
Intel quad core

**Comment by pingou67** [comment]
I see exactly the same high cpu usage when moving windows.

I have a nvidia graphic card with proprietary driver and cinnamon 2.8.4+rosa installed.

% uname -mrs
Linux 3.19.0-39-generic x86_64

**Comment by pingou67** [comment]
update: 
- it happens for all windows
- cinnamon is running in hardware rendering mode (I think because I don't have the software rendering message at login)

**Comment by pingou67** [comment]
Same after upgrade to linuxmint 17.3 with cinnamon 2.8.6+rosa

**Comment by willzyx15** [comment]
I have the same problem since I upgraded to linux mint 17.3. Cinnamon becomes slower when moving windows time after time and deleting several files. The cpu usage of cinnamon goes quite high when moving windows quickly. It happens for all windows. I have to restart cinnamon periodically to keep the system running smoothly.

Cinnamon 2.8.6 
kernel 4.2.0-19-generic 
nvidia graphic card is turned off, so it is running on intel card
intel core i5-2430m

**Comment by presianbg** [comment]
Same here. Moving windows objects in cinnamon require a lot of CPU. I'm on Fedora 23x64 bit. With latest mesa + optimus technology and latest cinnamon:

'''
cinnamon-settings-daemon-2.8.1-2.fc23.x86_64
cinnamon-control-center-filesystem-2.8.0-2.fc23.x86_64
cinnamon-translations-2.8.1-2.fc23.noarch
cinnamon-session-2.8.2-2.fc23.x86_64
cinnamon-desktop-2.8.0-2.fc23.x86_64
cinnamon-screensaver-2.8.0-2.fc23.x86_64
cinnamon-menus-2.8.0-2.fc23.x86_64
cinnamon-control-center-2.8.0-2.fc23.x86_64
imsettings-cinnamon-1.6.8-6.fc23.x86_64
cinnamon-2.8.6-1.fc23.x86_64

bumblebee-release-1.2-1.noarch
bumblebee-nonfree-release-1.2-1.noarch
bumblebee-nvidia-352.63-1.fc23.x86_64
bumblebee-3.2.1-9.fc23.x86_64

mesa-libwayland-egl-devel-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libwayland-egl-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libGLU-9.0.0-9.fc23.i686
mesa-libEGL-devel-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-filesystem-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libOSMesa-11.2.0-0.devel.18.af7ba98.fc23.i686
mesa-libgbm-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libgbm-11.2.0-0.devel.18.af7ba98.fc23.i686
mesa-libglapi-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libxatracker-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libGL-devel-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-dri-drivers-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-filesystem-11.2.0-0.devel.18.af7ba98.fc23.i686
mesa-dri-drivers-11.2.0-0.devel.18.af7ba98.fc23.i686
mesa-libGLU-9.0.0-9.fc23.x86_64
mesa-libEGL-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libEGL-11.2.0-0.devel.18.af7ba98.fc23.i686
mesa-libGLES-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libGL-11.2.0-0.devel.18.af7ba98.fc23.i686
mesa-libGL-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libGLU-devel-9.0.0-9.fc23.x86_64
mesa-libOSMesa-11.2.0-0.devel.18.af7ba98.fc23.x86_64
mesa-libglapi-11.2.0-0.devel.18.af7ba98.fc23.i686

'''

Best Regards,
Presian

**Comment by presianbg** [comment]
Just to be sure it is not mesa issue - same with downgraded (distro default) version: 

```
mesa-libOSMesa-11.0.6-1.20151122.fc23.x86_64
mesa-libGL-11.0.6-1.20151122.fc23.x86_64
mesa-libOSMesa-11.0.6-1.20151122.fc23.i686
mesa-libGLU-9.0.0-9.fc23.i686
mesa-libGLES-11.0.6-1.20151122.fc23.x86_64
mesa-libGL-11.0.6-1.20151122.fc23.i686
mesa-filesystem-11.0.6-1.20151122.fc23.i686
mesa-dri-drivers-11.0.6-1.20151122.fc23.x86_64
mesa-libGL-devel-11.0.6-1.20151122.fc23.x86_64
mesa-libglapi-11.0.6-1.20151122.fc23.x86_64
mesa-libgbm-11.0.6-1.20151122.fc23.x86_64
mesa-libEGL-11.0.6-1.20151122.fc23.i686
mesa-filesystem-11.0.6-1.20151122.fc23.x86_64
mesa-dri-drivers-11.0.6-1.20151122.fc23.i686
mesa-libGLU-9.0.0-9.fc23.x86_64
mesa-libglapi-11.0.6-1.20151122.fc23.i686
mesa-libEGL-devel-11.0.6-1.20151122.fc23.x86_64
mesa-libgbm-11.0.6-1.20151122.fc23.i686
mesa-libwayland-egl-11.0.6-1.20151122.fc23.x86_64
mesa-libwayland-egl-devel-11.0.6-1.20151122.fc23.x86_64
mesa-libGLU-devel-9.0.0-9.fc23.x86_64
mesa-libxatracker-11.0.6-1.20151122.fc23.x86_64
mesa-libEGL-11.0.6-1.20151122.fc23.x86_64

```

**Comment by presianbg** [comment]
Anything on this? My cinnamon is getting unusable :(

**Comment by brub2** [comment]
I'm affected by this issue since cinnamon 2.8 on Mint 17.2 (now updated to Mint 17.3). Cinnamon is running in hardware rendering mode. Moving windows becomes annoyingly hard (lag) to move after a few minutes of normal use and cinnamon has to be restarted. I have an nvidia GT 430. Update the kernel to 3.19 and all level 1 to 3 updates (including the latest proprietary driver) didn't fix anything. I can affect at least these applications: nemo, firefox, gedit, calc, evince, archive manager. I'm not sure but it didn't happen yet on nvidia-setting and banshee (but I don't play very much with these windows after they are open).

**Comment by presianbg** [comment]
Thunderbird and Chrome are making it worse.

**Comment by pingou67** [comment]
With chrome running, cinnamon is unusable.

**Comment by pingou67** [comment]
I've tested same computer with gnome-shell 3.16 and 3.18, no problem (even with chrome), moving windows is smooth with CPU below 10%. I say that because cinnamon is a fork of gnome-shell if that can help.

**Comment by ManIVIctorious** [comment]
Sounds like cinnamon is running in fallback mode. There can be various reasons for this but the only one that affected me personally was a libgl issue. The ones of you with nvidia graphics should try the proprietary driver and the nvidia-libgl

**Comment by presianbg** [comment]
@ManIVIctorious I'm not sure about this... I'm pretty sure that cinnamon is not in fallback mode. Also as I  shown, all related packages are up-to-date and beyond.

**Comment by brub2** [comment]
Of course nvidia proprietary driver is installed. If Cinnamon was running in fallback mode, I would have a warning message at login. 
Is this issue related to this? 
[https://devtalk.nvidia.com/default/topic/860367/linux/window-drag-lag-with-geforce-gtx-750-cinnamon-gnome-unity/3]
I didn't try the "solution" proposed in this topic yet.
I tried to manually reinitialize my cinnamon config (delete files), and then reactivate my couple of applets and extension and it didn't help. I'm now testing with a plain cinnamon config like preslanbg proposed. I have little hope, but we will see. More news soon.

**Comment by pingou67** [comment]
Same for me, nvidia-340-updates is installed and I haven't the warning message at login.

**Comment by brub2** [comment]
So far so good. Restoring Cinnamon to default (like presianbg suggested) seems to work. I will gradually reactivate my applets/extension; maybe one of them causes the issue.

**Comment by pingou67** [comment]
I have reinstalled two computers from scratch, one with nvidia 9600GT, the other with nvidia GTS450, both with proprietary drivers: same result, high CPU when moving windows and lag with chrome open. Like I said previously no problem with gnome-shell 3.16 and 3.18.

**Comment by JosephMcc** [comment]
The machine I have here has an older nvidia gtk260. I run the open source drivers on 3 installs, Mint, LMDE, and Ubuntu16.04 and don't have any of these issues. For those of you who have done new installs, do you get the issue when you first install or does it occur only after changing drivers? Or maybe after tweaking some settings or add some applets, extensions, or something else?

**Comment by pingou67** [comment]
As I have said previously, I have the problem with two new install of mint cinnamon 17.3, one with a nvidia 9600GT, the other with a nvidia GTS450, both with proprietary drivers and no tweaking (applets, extensions or other). When I move a windows, I see a high cpu usage with top. With chrome launched, cinnamon is unusable.

**Comment by JosephMcc** [comment]
@pingou67 which means at the very least you installed drivers and applications, so not a totally new install. Was it an issue when first installed? Using the open source drivers?

**Comment by pingou67** [comment]
@JosephMcc I just tested with the open source nouveau and same results. You do not see an high CPU usage (cinnamon process) when you move a window?

**Comment by JosephMcc** [comment]
No I don't and I've run cinnamon under multiple distros and with multiple machines. I mostly use firefox but I do use chrome on occasion and don't see an issue with it either. At worst my cpu usage will go to maybe 7 or 8% while moving a window but I don't think that is unexpected. That's why I was poking at people to find out what changes they made. There is some missing variable here that causes certain of you to run into this issue but not others.

Issues like this are nearly impossible to solve until we can find the cause and a way to reproduce.

**Comment by pingou67** [comment]
My cpu is about 40% when I move a window (more with nouveau). With gnome-shell (ubuntu 15.10), it is like you 7-8%.

I'm agree it's difficult to solve the problem without a way to reproduce it and I don't see the missing variable...

**Comment by pingou67** [comment]
Is there a way to verify that cinnamon doesn't running in software mode?

**Comment by brub2** [comment]
Maybe we are facing many related issues here. Even after resetting Cinnamon to defaults, the "top command in a terminal" still reveals a somewhat high (easily 20 to 40%) cpu for cinnamon process while moving a window. So maybe the CPU usage is too high (I didn't compare with other DE), but the main point for me is that moving has to remain smooth (it was not before resetting to defaults). 

Also, I have to conduct more tests but my issue (moving becomes not smooth after some time) reappeared after I activated the intellide panel. Maybe others can confirm this.

BTW I don't use chrome but sometimes I use chromium beta and it's not better or worse than with firefox.

**Comment by LucaDev** [comment]
I have the same problem, up to 100% load on one core when moving a window

**Comment by brycekottke** [comment]
same problem here. Weather it is moving a window, typing in a terminal, scrolling up and down on a webpage. I get a 1 second to 2 second freeze / lag every time the CPU randomly spikes for cinnamon. see below the random 81.2 and 99.5 CPU spike for the cinnamon process.

**Process CPU Spike:**
```
~ $ top -d 1 |grep cinnamon
 3605 jim     20   0 2158260 579332 105884 S   1.0  3.5  79:02.91 cinnamon                                          
 3605 jim     20   0 2158260 579332 105884 S   1.0  3.5  79:02.92 cinnamon                                          
 3605 jim     20   0 2158260 579332 105884 S   1.0  3.5  79:02.93 cinnamon                                          
 3605 jim     20   0 2158260 579332 105884 S   1.0  3.5  79:02.94 cinnamon                                          
 3605 jim     20   0 2158260 579332 105884 S   1.0  3.5  79:02.95 cinnamon                                          
 3605 jim     20   0 2158260 579332 105884 S   1.0  3.5  79:02.96 cinnamon                                          
 3605 jim     20   0 2158324 579352 105884 R  81.2  3.5  79:03.78 cinnamon                                          
 3605 jim     20   0 2158388 579352 105884 R  99.5  3.5  79:04.78 cinnamon                                          
 3605 jim     20   0 2158388 579452 105884 S   9.0  3.5  79:04.87 cinnamon                                          
 3605 jim     20   0 2158388 579452 105884 S   1.0  3.5  79:04.88 cinnamon
```

**Cinnamon Version:**
```
~ $ cinnamon --version
Cinnamon 2.8.8
```

**CPU Info:**
```
~ $ head -15 /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 42
model name	: Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz
stepping	: 7
microcode	: 0x29
cpu MHz		: 1601.054
cache size	: 8192 KB
physical id	: 0
siblings	: 8
core id		: 0
cpu cores	: 4
```

**Graphics Card Info:**
```
~ $ lspci -v |grep -A 15 VGA
01:00.0 VGA compatible controller: NVIDIA Corporation GM107 [GeForce GTX 750 Ti] (rev a2) (prog-if 00 [VGA controller])
	Subsystem: Gigabyte Technology Co., Ltd Device 36d1
	Flags: bus master, fast devsel, latency 0, IRQ 30
	Memory at fa000000 (32-bit, non-prefetchable) [size=16M]
	Memory at c0000000 (64-bit, prefetchable) [size=256M]
	Memory at d0000000 (64-bit, prefetchable) [size=32M]
	I/O ports at e000 [size=128]
	[virtual] Expansion ROM at fb000000 [disabled] [size=512K]
	Capabilities: <access denied>
	Kernel driver in use: nvidia

```

**Nvidia Drivers Installed:** 

  - currently using nvidia-352 driver.

```
~ $ dpkg -l |grep nvidia-
rc  nvidia-304                                  304.131-0ubuntu0.14.04.1                            amd64        NVIDIA legacy binary driver - version 304.131
rc  nvidia-340                                  340.96-0ubuntu0.14.04.1                             amd64        NVIDIA binary driver - version 340.96
ii  nvidia-352                                  352.63-0ubuntu0.14.04.1                             amd64        NVIDIA binary driver - version 352.63
ii  nvidia-common                               1:0.2.91.11linuxmint2                               amd64        transitional package for ubuntu-drivers-common
rc  nvidia-libopencl1-340                       340.96-0ubuntu0.14.04.1                             amd64        NVIDIA OpenCL Driver and ICD Loader library
rc  nvidia-opencl-icd-340                       340.96-0ubuntu0.14.04.1                             amd64        NVIDIA OpenCL ICD
ii  nvidia-opencl-icd-352                       352.63-0ubuntu0.14.04.1                             amd64        NVIDIA OpenCL ICD
ii  nvidia-prime                                0.6.2linuxmint1                                     amd64        Tools to enable NVIDIA's Prime
ii  nvidia-settings                             331.20-0ubuntu8                                     amd64        Tool for configuring the NVIDIA graphics driver
```

**Temporary Resolution:**

  - reboot machine.

this tends to resolve it for a temporary period of time. But just comes right back again.

**Comment by catb0t** [comment]
Same issue. 

**Example**
```
cat @ mint-kitty : ~ $ top -d 1 | grep cinnamon
24436 cat       20   0 2004656 260496  56344 S  18.8  4.3   9:08.08 cinnamon
24436 cat       20   0 2004656 260496  56344 S  14.9  4.3   9:08.23 cinnamon
24436 cat       20   0 2004656 260496  56344 S  13.9  4.3   9:08.37 cinnamon
24298 cat       20   0 1235456 104840  38368 S   1.0  1.7   0:21.70 cinnamon-settin
24436 cat       20   0 2004656 260496  56344 S  13.9  4.3   9:08.51 cinnamon
24436 cat       20   0 2004720 260504  56344 S  15.0  4.3   9:08.66 cinnamon
24436 cat       20   0 2004720 260528  56344 R  14.9  4.3   9:08.81 cinnamon
24436 cat       20   0 2004720 260532  56344 S  16.8  4.3   9:08.98 cinnamon
24436 cat       20   0 2004720 260536  56344 S  31.7  4.3   9:09.30 cinnamon
24298 cat       20   0 1235456 104840  38368 S   1.0  1.7   0:21.71 cinnamon-settin
24436 cat       20   0 2004720 260536  56344 S  31.7  4.3   9:09.62 cinnamon
24298 cat       20   0 1235456 104840  38368 S   1.0  1.7   0:21.72 cinnamon-settin
24436 cat       20   0 2004720 260536  56344 S  39.0  4.3   9:10.01 cinnamon
24298 cat       20   0 1235456 104840  38368 S   2.0  1.7   0:21.74 cinnamon-settin
24436 cat       20   0 2004720 260536  56344 S  24.8  4.3   9:10.26 cinnamon
24436 cat       20   0 2004788 260536  56344 S  36.6  4.3   9:10.63 cinnamon
24298 cat       20   0 1235456 104840  38368 S   2.0  1.7   0:21.76 cinnamon-settin
24436 cat       20   0 2004788 260596  56344 S  30.7  4.3   9:10.94 cinnamon
24436 cat       20   0 2006084 260596  56344 R  75.2  4.3   9:11.70 cinnamon
24298 cat       20   0 1235456 104840  38368 S   1.0  1.7   0:21.77 cinnamon-settin
24436 cat       20   0 2003724 260356  56100 R  90.1  4.3   9:12.61 cinnamon
24298 cat       20   0 1235456 104868  38368 R  16.8  1.7   0:21.94 cinnamon-settin
24436 cat       20   0 2016516 272692  56040 R  93.1  4.5   9:13.55 cinnamon
24298 cat       20   0 1235456 104868  38368 S  38.6  1.7   0:22.33 cinnamon-settin
24436 cat       20   0 2033140 289348  56036 R  91.1  4.8   9:14.47 cinnamon
24298 cat       20   0 1235456 104868  38368 S   1.0  1.7   0:22.34 cinnamon-settin
24436 cat       20   0 2044008 300408  56036 R  76.0  5.0   9:15.23 cinnamon
24298 cat       20   0 1235456 104868  38368 S   1.0  1.7   0:22.35 cinnamon-settin
24436 cat       20   0 2053620 309412  56064 R  64.4  5.1   9:15.88 cinnamon
24298 cat       20   0 1235456 104868  38368 S   1.0  1.7   0:22.36 cinnamon-settin
24436 cat       20   0 2058652 313048  56064 S  59.4  5.2   9:16.48 cinnamon
24436 cat       20   0 2061216 314880  56068 S  52.5  5.2   9:17.01 cinnamon
24298 cat       20   0 1235456 104868  38368 S   1.0  1.7   0:22.37 cinnamon-settin
```

**Version**
```
130 | cat @ mint-kitty : ~ $ cinnamon --version
Cinnamon 2.8.6
```

**CPU**
```
cat @ mint-kitty : ~ $ head -15 /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 42
model name	: Intel(R) Core(TM) i7-2670QM CPU @ 2.20GHz
stepping	: 7
microcode	: 0x29
cpu MHz		: 2348.414
cache size	: 6144 KB
physical id	: 0
siblings	: 8
core id		: 0
cpu cores	: 4
apicid		: 0
initial apicid	: 0

cat @ mint-kitty : ~ $ lspci -v |grep -A 15 VGA
00:02.0 VGA compatible controller: Intel Corporation 2nd Generation Core Processor Family Integrated Graphics Control
ler (rev 09) (prog-if 00 [VGA controller])
	DeviceName:  Onboard IGD
	Subsystem: Toshiba America Info Systems 2nd Generation Core Processor Family Integrated Graphics Controller
	Flags: bus master, fast devsel, latency 0, IRQ 34
	Memory at f7800000 (64-bit, non-prefetchable) [size=4M]
	Memory at e0000000 (64-bit, prefetchable) [size=256M]
	I/O ports at f000 [size=64]
	Expansion ROM at <unassigned> [disabled]
	Capabilities: <access denied>
	Kernel driver in use: i915
	Kernel modules: i915

00:16.0 Communication controller: Intel Corporation 6 Series/C200 Series Chipset Family MEI Controller #1 (rev 04)
	Subsystem: Toshiba America Info Systems 6 Series/C200 Series Chipset Family MEI Controller
	Flags: bus master, fast devsel, latency 0, IRQ 35
	Memory at f7f0a000 (64-bit, non-prefetchable) [size=16]
```

**Drivers**
```cat @ mint-kitty : ~ $ dpkg -l | grep mesa
ii  libegl1-mesa:amd64                                          11.2.0-1ubuntu2                                    amd64        free implementation of the EGL API -- runtime
ii  libegl1-mesa-dev:amd64                                      11.2.0-1ubuntu2                                    amd64        free implementation of the EGL API -- development files
ii  libgl1-mesa-dev:amd64                                       11.2.0-1ubuntu2                                    amd64        free implementation of the OpenGL API -- GLX development files
ii  libgl1-mesa-dev:i386                                        11.2.0-1ubuntu2                                    i386         free implementation of the OpenGL API -- GLX development files
ii  libgl1-mesa-dri:amd64                                       11.2.0-1ubuntu2                                    amd64        free implementation of the OpenGL API -- DRI modules
ii  libgl1-mesa-dri:i386                                        11.2.0-1ubuntu2                                    i386         free implementation of the OpenGL API -- DRI modules
ii  libgl1-mesa-glx:amd64                                       11.2.0-1ubuntu2                                    amd64        free implementation of the OpenGL API -- GLX runtime
ii  libgl1-mesa-glx:i386                                        11.2.0-1ubuntu2                                    i386         free implementation of the OpenGL API -- GLX runtime
ii  libglapi-mesa:amd64                                         11.2.0-1ubuntu2                                    amd64        free implementation of the GL API -- shared library
ii  libglapi-mesa:i386                                          11.2.0-1ubuntu2                                    i386         free implementation of the GL API -- shared library
ii  libgles1-mesa:amd64                                         11.2.0-1ubuntu2                                    amd64        free implementation of the OpenGL|ES 1.x API -- runtime
ii  libgles2-mesa:amd64                                         11.2.0-1ubuntu2                                    amd64        free implementation of the OpenGL|ES 2.x API -- runtime
ii  libgles2-mesa-dev:amd64                                     11.2.0-1ubuntu2                                    amd64        free implementation of the OpenGL|ES 2.x API -- development files
ii  libglu1-mesa:amd64                                          9.0.0-2.1                                          amd64        Mesa OpenGL utility library (GLU)
ii  libglu1-mesa:i386                                           9.0.0-2.1                                          i386         Mesa OpenGL utility library (GLU)
ii  libglu1-mesa-dev:amd64                                      9.0.0-2.1                                          amd64        Mesa OpenGL utility library -- development files
ii  libosmesa6:amd64                                            11.2.0-1ubuntu2                                    amd64        Mesa Off-screen rendering extension
ii  libosmesa6:i386                                             11.2.0-1ubuntu2                                    i386         Mesa Off-screen rendering extension
ii  libosmesa6-dev:amd64                                        11.2.0-1ubuntu2                                    amd64        Mesa Off-screen rendering extension -- development files
ii  libwayland-egl1-mesa:amd64                                  11.2.0-1ubuntu2                                    amd64        implementation of the Wayland EGL platform -- runtime
ii  mesa-common-dev:amd64                                       11.2.0-1ubuntu2                                    amd64        Developer documentation for Mesa
ii  mesa-common-dev:i386                                        11.2.0-1ubuntu2                                    i386         Developer documentation for Mesa
ii  mesa-utils                                                  8.3.0-1                                            amd64        Miscellaneous Mesa GL utilities
ii  mesa-vdpau-drivers:amd64                                    11.2.0-1ubuntu2                                    amd64        Mesa VDPAU video acceleration drivers

cat @ mint-kitty : ~ $ dpkg -l | grep intel
ii  intel-microcode                                             3.20151106.1                                       amd64        Processor microcode firmware for Intel CPUs
ii  libdrm-intel1:amd64                                         2.4.67-1ubuntu0.16.04.1                            amd64        Userspace interface to intel-specific kernel DRM services -- runtime
ii  libdrm-intel1:i386                                          2.4.67-1ubuntu0.16.04.1                            i386         Userspace interface to intel-specific kernel DRM services -- runtime
ii  libpresage-data                                             0.9.1-1.1ubuntu2                                   all          intelligent predictive text entry platform (data files)
rc  libpresage1:amd64                                           0.9.1-0ubuntu2                                     amd64        intelligent predictive text entry platform (shared library)
ii  libpresage1v5:amd64                                         0.9.1-1.1ubuntu2                                   amd64        intelligent predictive text entry platform (shared library)
ii  pyprompter                                                  0.9.1-1.1ubuntu2                                   all          intelligent predictive wxPython text editor
ii  python-presage                                              0.9.1-1.1ubuntu2                                   amd64        intelligent predictive text entry platform (Python binding)
ii  whois                                                       5.2.11                                             amd64        intelligent WHOIS client
ii  wvdial                                                      1.61-4.1                                           amd64        intelligent Point-to-Point Protocol dialer
ii  xserver-xorg-video-intel                                    2:2.99.917+git20160325-1ubuntu1                    amd64        X.Org X server -- Intel i8xx, i9xx display driver
```
Rebooting or completely obliterating Cinnamon fixes it until Cinnamon is up for around 20 minutes, then the problem starts happening.

**Comment by kolorafa** [comment]
How can someone verify that cinnamon is not in software render mode?
I did a test by loggin out and in to "Cinnamon (software)" and didn't see any difference, the same cpu usage and the same cpu usage behavior.

>glxinfo |grep render 
>direct rendering: Yes

When on "idle" just with (10+) opened windows chrome, terminals etc:

>$ top -d 1 | grep cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  29,4 12,3   2925:48 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  21,6 12,3   2925:48 R cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  27,5 12,3   2925:48 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  24,5 12,3   2925:48 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  29,4 12,3   2925:49 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  26,5 12,3   2925:49 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  25,7 12,3   2925:49 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  31,1 12,3   2925:49 R cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  21,8 12,3   2925:50 R cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  29,4 12,3   2925:50 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  23,5 12,3   2925:50 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  19,6 12,3   2925:50 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  21,6 12,3   2925:51 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  23,5 12,3   2925:51 S cinnamon
> 1058 kolorafa  20   0 2773,8m 1,445g  24,5 12,3   2925:51 S cinnamon

When moving any window:

>$ top -d 1 | grep cinnamon
> 1058 kolorafa  20   0 2774,0m 1,445g  64,7 12,3   2926:07 R cinnamon
> 1058 kolorafa  20   0 2774,0m 1,445g  21,6 12,3   2926:07 S cinnamon
> 1058 kolorafa  20   0 2774,0m 1,445g  45,6 12,3   2926:08 R cinnamon
> 874 kolorafa  20   0 1021,3m  19,6m   1,0  0,2   4:34.83 S cinnamon-settin
> 1058 kolorafa  20   0 2774,0m 1,445g  69,6 12,3   2926:08 S cinnamon
> 1058 kolorafa  20   0 2774,0m 1,445g  79,2 12,3   2926:09 R cinnamon
> 1058 kolorafa  20   0 2773,9m 1,446g  87,3 12,3   2926:10 R cinnamon
> 1058 kolorafa  20   0 2773,9m 1,446g  76,5 12,3   2926:11 R cinnamon
> 1058 kolorafa  20   0 2773,9m 1,446g  76,5 12,3   2926:12 R cinnamon
> 1058 kolorafa  20   0 2773,9m 1,446g  77,2 12,3   2926:13 R cinnamon
>  874 kolorafa  20   0 1021,3m  19,6m   1,0  0,2   4:34.84 S cinnamon-settin
> 1058 kolorafa  20   0 2773,9m 1,446g  80,4 12,3   2926:13 R cinnamon
> 1058 kolorafa  20   0 2774,0m 1,446g  79,4 12,3   2926:14 R cinnamon
> 1058 kolorafa  20   0 2774,0m 1,446g  81,4 12,3   2926:15 R cinnamon
> 1058 kolorafa  20   0 2774,0m 1,446g  82,4 12,3   2926:16 R cinnamon
> 1058 kolorafa  20   0 2774,0m 1,446g  74,5 12,3   2926:17 R cinnamon
> 1058 kolorafa  20   0 2774,0m 1,446g  77,5 12,3   2926:17 R cinnamon
> 1058 kolorafa  20   0 2773,9m 1,445g  78,4 12,3   2926:19 R cinnamon
> 1058 kolorafa  20   0 2773,9m 1,445g  76,5 12,3   2926:20 R cinnamon


when running pavucontrol on an active workspace (only when on active the cpu spikes):

>$ top -d 1 | grep cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  88,2 12,3   2927:42 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  91,1 12,3   2927:43 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  86,3 12,3   2927:44 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  91,2 12,3   2927:45 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  89,2 12,3   2927:46 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  87,3 12,3   2927:47 S cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  92,2 12,3   2927:48 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  90,2 12,3   2927:49 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  89,2 12,3   2927:49 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  90,2 12,3   2927:50 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  90,2 12,3   2927:51 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  85,3 12,3   2927:52 R cinnamon
> 1058 kolorafa  20   0 2774,1m 1,446g  88,3 12,3   2927:53 R cinnamon

While running glxgears:
>$ glxgears
>Running synchronized to the vertical refresh.  The framerate should be
>approximately the same as the monitor refresh rate.
>297 frames in 5.0 seconds = 59.233 FPS
>301 frames in 5.0 seconds = 60.025 FPS
>301 frames in 5.0 seconds = 60.017 FPS
>^C


> 1058 kolorafa  20   0 2785,3m 1,450g  93,1 12,4   2946:21 R cinnamon
> 1058 kolorafa  20   0 2785,3m 1,450g  95,1 12,4   2946:22 R cinnamon
> 1058 kolorafa  20   0 2785,2m 1,450g 104,9 12,4   2946:23 R cinnamon
> 1058 kolorafa  20   0 2785,2m 1,450g  94,1 12,4   2946:24 R cinnamon
> 1058 kolorafa  20   0 2785,2m 1,450g  92,2 12,4   2946:25 R cinnamon
> 1058 kolorafa  20   0 2785,2m 1,450g  94,1 12,4   2946:26 R cinnamon
> 1058 kolorafa  20   0 2785,2m 1,450g  69,6 12,4   2946:27 R cinnamon

OS: Arch Antergos
Graphic: GTX 260
Cpu: 2x Intel(R) Xeon(R) CPU            5140  @ 2.33GHz
HP Workstation xv6400
Cinnamon 3.0.7


On gnome i had problem with games fps, like 5fps...
After installing cinnamon (without reinstalling) i get easily 50fps, but the overall experience is cpu-hungry and for now (11 days uptime 24h/day) somehow it's starting to get to me that cinnamon probably works in software rendering mode and i don't know what to test/do other that trying to reinstall or switch to something else, but i really like cinnamon interface, more than gnome.

I would try Mint, but In the past i did try linux and did go back to windows because somehow every distro upgrade ended in failure ... And after years failing with disto upgrades on debian servers i somehow got burned by 'frozen' distros... 

Maybe someone could recommend some rolling distro with best Cinnamon support?
Or how to debug why cinnamon uses so much from my (old/ancient) cpu.

**Comment by mainmachine** [comment]
@kolorafa - Manjaro with Cinnamon is pretty great. Couldn't wrap my head around pacman and the AUR, so I went back to Ubuntu as a base pretty fast, but it was easy to set up and worked quite well.

**Comment by JosephMcc** [comment]
@kolorafa I actually have the exact same video card that you do. I don't see those issues with Cinnamon. I do have a Core I7 instead of the Xeons but moving windows will maybe cause my cpu usage to 5%. I've had this machine for quite some time and will say that the Nvidia drivers have not been great. My system is far more stable on the open source driver.

**Comment by kolorafa** [comment]
I was looking into memory usage, (as i did hit 12GB usage) while closing chrome and some random apps (like nemo that did use 3GB ram) the last thing was cinnamon that did use 1.5GB of ram.

So i did ALT+F2 "r" ENTER
cinnamon did restart and now it uses 390MB 
idle cpu went down from 25% to 5%
moving any window did go down from 95% to 40%
pavucontrol from 95% to 20%
glxgears from 95% to 19%

It's not a problem to restart cinnamon every few days, as you don't lose any windows, all running stuff still run, but i hope eventually it will get fixed.


-----
So for people that encounter this problem, try to restart cinnamon as a temporary fix

alt+f2, type r and enter

**Comment by JosephMcc** [comment]
@kolorafa We have had other people report high memory usage from Cinnamon that none of us can manage to reproduce. Since we use the same hardware I would be curious if you would mind providing some more info. For me, using alt+f2 and r, restarting cinnamon gives me around 180mb memory usage. After an uptime of a couple days it never goes above 210.

I would be curious
- Which driver do you use? Nvidia or open source
- How many panels?
- Could you provide a list of applets, desklets, and extensions you use?
- Do you have any applications that you always have running in the background. Any special applications you run startup?

The info would be appreciated so I could try to reproduce that level of memory usage.

**Comment by kolorafa** [comment]
- Nvidia prop drivers (also ati card in pci slot, but can't use both in linux)
- 2 screens - 2 panels, primary standard stuff, secondery with window list and multi-core system monitor at 500ms refresh rate cpu bars - is there a way to export the panel list/settings?
http://kolorafa.firehost.pl/test/snapshot-2016-09-09-23-12-56.png
http://kolorafa.firehost.pl/test/snapshot-2016-09-09-23-16-40.png

- apps that i use ... a lot of them:
- skype
- nagstamon
- teamviewer
- dropbox
- pidgin
- quassel client
- synergy (remote mouse)
- teamspeak3
- coffeine for few days
- syncthing - with invisible icon - because it didn't have enough delay at start like other apps
- yakyak - hangouts client
- nomachine - but i can uninstall it, i dont use it
- hotshots - screenshots
- audacious
- kadu - polish communicator gadu-gadu
- chrome
- timedoctor - sometimes for work - probably no one use it


Bugs that i encounter:
1. half of the tray icons show menu 20px higher than mouse 

- but that probably because my primary monitor is on right, and the left 0,0 screen is smaller and it move menu up so it would "show" on the left screen resolution

2. and i switch workspaces a lot, and there is 1 clearly visible bug with switching workspaces
somehow only the 'kadu' app have strange "going down" problem, every time i switch workspace the window magically goes down, the problem was also in gnome but did show differently
in gnome the window did really go down, visible
in cinnamon when i switch workspaces the window still renders in the same position but when you move mouse over it for example over borders or any buttons it is down, it still works but to click a button or move the window you need to touch X px below, every time i change workspace it goes lower

- the problem is only with this one app

cinnamon now use 452,2m , but i have more windows

I could provide access to my machine if that would help gathering data.

**Comment by jezekus** [comment]
Hello everybody, I'm experiencing the quite same problem on my Lenovo T440.
Even there is nothing moving on the screen `cinnamon --replace` process is "ticking" up to 10% CPU on regular timeframe like 1 sec.
When there is something - window move, opening new windows or video playing this process is going CPU high up to 70%

I'm concerned about it as  it drains the battery and won't allow idling of CPU during office work -writing emails, documents, im etc. Which should not need too  much CPU power.

Running up to date Fedora 24 

> uname -a
Linux nb1697 4.8.6-201.fc24.x86_64 #1 SMP Thu Nov 3 14:38:57 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

> [jezek@nb1697 ~]$ cinnamon --version
Cinnamon 3.0.7


>[jezek@nb1697 ~]$ head -15 /proc/cpuinfo
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 69
model name	: Intel(R) Core(TM) i5-4300U CPU @ 1.90GHz
stepping	: 1
microcode	: 0x1f
cpu MHz		: 1908.416
cache size	: 3072 KB
physical id	: 0
siblings	: 4
core id		: 0
cpu cores	: 2
apicid		: 0
initial apicid	: 0

> [jezek@nb1697 ~]$ lspci -v |grep -A 15 VGA
00:02.0 VGA compatible controller: Intel Corporation Haswell-ULT Integrated Graphics Controller (rev 0b) (prog-if 00 [VGA controller])
	Subsystem: Lenovo Device 220c
	Flags: bus master, fast devsel, latency 0, IRQ 46
	Memory at f0000000 (64-bit, non-prefetchable) [size=4M]
	Memory at e0000000 (64-bit, prefetchable) [size=256M]
	I/O ports at 3000 [size=64]
	[virtual] Expansion ROM at 000c0000 [disabled] [size=128K]
	Capabilities: <access denied>
	Kernel driver in use: i915
	Kernel modules: i915


Is there anything I could help with ? Logs? Tests?

**Comment by leigh123linux** [comment]
@jezekus 

I'm not seeing this issue with my T410 with F24 cinnamon-3.0.x.
I'm no longer using F24 on it as I have moved onto F25 cinnamon-3.2.x (still no seeing this issue).

F24 & F25 wont be getting any more 3.0.x updates

https://copr.fedorainfracloud.org/coprs/leigh123linux/cinnamon-next/

**Comment by genail** [comment]
I can confirm that this issue is still in effect on Mint 18 with Cinnamon 3.0.7.

> uname -a
> Linux highwind 4.4.0-21-generic #37-Ubuntu SMP Mon Apr 18 18:33:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
> 
> head -15 /proc/cpuinfo
> processor	: 0
> vendor_id	: AuthenticAMD
> cpu family	: 16
> model		: 10
> model name	: AMD Phenom(tm) II X6 1075T Processor
> stepping	: 0
> microcode	: 0x10000bf
> cpu MHz		: 800.000
> cache size	: 512 KB
> physical id	: 0
> siblings	: 6
> core id		: 0
> cpu cores	: 6
> apicid		: 0
> initial apicid	: 0
> 
> lspci -v |grep -A 15 VGA
> 	Flags: bus master, VGA palette snoop, 66MHz, medium devsel, latency 64
> 	Bus: primary=00, secondary=03, subordinate=03, sec-latency=64
> 	I/O behind bridge: 0000c000-0000cfff
> 	Memory behind bridge: fde00000-fdefffff
> 	Prefetchable memory behind bridge: fdd00000-fddfffff
> 
> 00:14.5 USB controller: Advanced Micro Devices, Inc. [AMD/ATI] SB7x0/SB8x0/SB9x0 USB OHCI2 Controller (prog-if 10 [OHCI])
> 	Subsystem: Gigabyte Technology Co., Ltd SB7x0/SB8x0/SB9x0 USB OHCI2 Controller
> 	Flags: bus master, 66MHz, medium devsel, latency 32, IRQ 18
> 	Memory at fe028000 (32-bit, non-prefetchable) [size=4K]
> 	Kernel driver in use: ohci-pci
> 
> 00:18.0 Host bridge: Advanced Micro Devices, Inc. [AMD] Family 10h Processor HyperTransport Configuration
> 	Flags: fast devsel
> 	Capabilities: <access denied>
> 
> --
> 01:00.0 VGA compatible controller: NVIDIA Corporation GK107 [GeForce GTX 650] (rev a1) (prog-if 00 [VGA controller])
> 	Subsystem: Micro-Star International Co., Ltd. [MSI] GK107 [GeForce GTX 650]
> 	Flags: bus master, fast devsel, latency 0, IRQ 27
> 	Memory at fb000000 (32-bit, non-prefetchable) [size=16M]
> 	Memory at c0000000 (64-bit, prefetchable) [size=256M]
> 	Memory at de000000 (64-bit, prefetchable) [size=32M]
> 	I/O ports at ef00 [size=128]
> 	[virtual] Expansion ROM at fc000000 [disabled] [size=512K]
> 	Capabilities: <access denied>
> 	Kernel driver in use: nvidia
> 	Kernel modules: nvidiafb, nouveau, nvidia_367, nvidia_367_drm
> 
> 01:00.1 Audio device: NVIDIA Corporation GK107 HDMI Audio Controller (rev a1)
> 	Subsystem: Micro-Star International Co., Ltd. [MSI] GK107 HDMI Audio Controller
> 	Flags: bus master, fast devsel, latency 0, IRQ 19
> 	Memory at fcffc000 (32-bit, non-prefetchable) [size=16K]

**Comment by christophra** [comment]
I've had this issue of high CPU usage when moving windows as well. For me it was exclusively associated with the applet "sticky@scollins".

**Comment by ACinnamonUserInUK** [comment]
I have this problem (on a laptop with, only, inbuilt Intel graphics), having had to reinstall Cinnamon. I notice that in the 'Melange' debugger log I have this report: 'Cinnamon Software Rendering mode enabled'. However, I receive no other notice that that mode is enabled.

I have trouble enabling the hardware rendering. Reinstalling my Intel graphics drivers didn't work. Nor did

sudo update-rc.d -f mdm remove
sudo update-rc.d mdm defaults

Now, resetting Cinnamon settings - via 'troubleshooting' (via right-clicking the panel) did work, once, and only until I rebooted; and now doing that reset seems not to help at all . .

**Comment by ACinnamonUserInUK** [comment]
The following may be worth noting, both

(1) perhaps as a means to improve Cinnamon (and I do think that it should _always_ display the 'software rendering warning' upon login, if such rendering is being used; the problem should not, on occasion, be hidden in a log)
and
(2) to help others having problems with Cinnamon as Cinnamon is at present.

I restored my whole root partition from an image backup - and the problem persisted.

I restored my whole _home_ partition from an image backup - _and the problem was solved_.

**Comment by leigh123linux** [comment]
Cinnamon 3.0.x is EOL, if you can reproduce with cinnamon 3.8.x feel free to file a new issue.
